src/OrbitGl/TimelineUiTest.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include "GlCanvas.h"
+#include "MockBatcher.h"
+#include "MockTextRenderer.h"
+#include "MockTimelineInfo.h"
+#include "TimelineUi.h"
+#include "Viewport.h"
+
+namespace orbit_gl {
+
+class TimelineUiTest : public TimelineUi {
+ public:
+  TimelineUiTest(int width)
+      : TimelineUi(nullptr /*parent*/, &mock_timeline_info, &viewport_, &layout_),
+        primitive_assembler_(&mock_batcher_),
+        viewport_(width, 0) {
+    mock_timeline_info.SetWorldWidth(width);
+  }
+
+  void TestUpdatePrimitives(uint64_t min_tick, uint64_t max_tick, bool is_small_screen = false) {
```

Comment:
The case where `is_small_screen` is true is not tested in the PR right? Either add a test, or remove the option?

Reply:
I would prefer to have it like this. I'm not having the test because the condition between labels and major_ticks that eventually I would like to solve. After that I would like to have a Big screen test, a medium screen test and a small screen test.

Reply:
Well that's not really an argument. You can add that when needed.

Reply:
Maybe I am not being clear. The statement `EXPECT_GE(num_major_ticks, 2);` is true only if the screen is big enough to show 2 labels. This is actually one of the most discussed points and we aligned in that statement with Carsten. 

The alternative is erasing this variable and NOT checking this condition, but I prefer to check this condition, because in general the screen is big enough to show 2 labels and it will catch a lot of silly mistakes.

Reply:
Oh I see. Sry. I misinterpreted Pierric's comment I guess.

Reply:
I will wait for Pierric's review anyways. Maybe I'm misinterpretating something.

We are prioritizing normal behavior here, and only trying to be consistent in extreme cases, so I would like to test as most as possible in this visual tests. And I found this boolean "is_small_screen" a good trade-off.

Reply:
Maybe just also add a small screen test in this PR?

Reply:
We are having an issue (listed below) about comparing major ticks and labels. It's listed below. I can test a small screen without comparing that and listing the bug there. What do you think?

Reply:
Ok. I modified and we are now testing both a big screen and a tiny one. Also I erase that variable and we are getting the boolean variable internally. I think I've addressed everything. Let me know what you think.